src/core/etl/src/Flow/ETL/Row/Entry/UuidEntry.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Row\Entry;
+
+use Flow\ETL\Exception\InvalidArgumentException;
+use Flow\ETL\Exception\RuntimeException;
+use Flow\ETL\Row\Entry;
+use Flow\ETL\Row\Reference;
+use Flow\ETL\Row\Schema\Definition;
+use Ramsey\Uuid\Uuid as RamseyUuid;
+use Ramsey\Uuid\UuidInterface as RamseyUuidInterface;
+use Symfony\Component\Uid\Uuid as SymfonyUuid;
+
+/**
+ * @implements Entry<RamseyUuidInterface|SymfonyUuid, array{name: string, value: string}>
+ */
+final class UuidEntry implements \Stringable, Entry
+{
+    use EntryRef;
+
+    private readonly RamseyUuidInterface|SymfonyUuid $value;
+
+    /**
+     * @throws InvalidArgumentException|RuntimeException
+     */
+    public function __construct(private readonly string $name, string|RamseyUuidInterface|SymfonyUuid $value)
+    {
+        if ('' === $name) {
+            throw InvalidArgumentException::because('Entry name cannot be empty');
+        }
+
+        if (\is_string($value)) {
+            if (\class_exists(RamseyUuid::class)) {
```

Comment:
I think this will throw an error when class does not exists, afaik you should use FQCN string instead here.

Reply:
Right, fixed.